external-crates/move/crates/move-analyzer/trace-debug/src/extension.ts

```diff
@@ -221,42 +232,47 @@ class MoveConfigurationProvider implements vscode.DebugConfigurationProvider {
-    const pkgRoot = await findPkgRoot(editor.document.uri.fsPath);
-    if (!pkgRoot) {
-        throw new Error(`Cannot find package root for file  '${editor.document.uri.fsPath}'`);
-    }
-
+    const openedFilePath = editor.document.uri.fsPath;
+    const openedFileExt = path.extname(openedFilePath);
     let tracedFunctions: string[] = [];
-    if (path.extname(editor.document.uri.fsPath) === MOVE_FILE_EXT) {
-        const pkgModules = findSrcModules(editor.document.getText());
-        if (pkgModules.length === 0) {
-            throw new Error(`Cannot find any modules in file '${editor.document.uri.fsPath}'`);
-        }
-        tracedFunctions = findTracedFunctionsFromPath(pkgRoot, pkgModules);
-    } else if (path.extname(editor.document.uri.fsPath) === BCODE_FILE_EXT) {
-        const modulePattern = /\bmodule\s+\d+\.\w+\b/g;
-        const moduleSequences = editor.document.getText().match(modulePattern);
-        if (!moduleSequences || moduleSequences.length === 0) {
-            throw new Error(`Cannot find module declaration in disassembly file '${editor.document.uri.fsPath}'`);
-        }
-        // there should be only one module declaration in a disassembly file
-        const [pkgAddrStr, module] = moduleSequences[0].substring('module'.length).trim().split('.');
-        const pkgAddr = parseInt(pkgAddrStr);
-        if (isNaN(pkgAddr)) {
-            throw new Error(`Cannot parse package address from '${pkgAddrStr}' in disassembly file '${editor.document.uri.fsPath}'`);
-        }
-        tracedFunctions = findTracedFunctionsFromTrace(pkgRoot, pkgAddr, module);
-    } else {
-        // this is a JSON (hopefully) trace as this function is only called if
-        // the active file is either a .move, .mvb, or .json file
+    if (openedFileExt === JSON_FILE_EXT) {
+        // this is hopefully a trace in JSON format (we do check if JSON schema
+        // is correct in the following code)
         const fpath = editor.document.uri.fsPath;
         const tracedFunctionInfo = getTracedFunctionInfo(fpath);
         tracedFunctions = [constructTraceInfo(fpath, tracedFunctionInfo)];
-    }
-    if (!tracedFunctions || tracedFunctions.length === 0) {
-        throw new Error(`No traced functions found for package at '${pkgRoot}'`);
-    }
+    } else {
+        const pkgRoot = await findPkgRoot(editor.document.uri.fsPath);
+        if (!pkgRoot) {
+            throw new Error(`Cannot find package root for file  '${editor.document.uri.fsPath}'`);
+        }
+        if (path.extname(editor.document.uri.fsPath) === MOVE_FILE_EXT) {
+            const pkgModules = findSrcModules(editor.document.getText());
+            if (pkgModules.length === 0) {
+                throw new Error(`Cannot find any modules in file '${editor.document.uri.fsPath}'`);
+            }
+            tracedFunctions = findTracedFunctionsFromPath(pkgRoot, pkgModules);
+        } else {
```

Comment:
Might be nice to still check for the `BCODE_FILE_EXT` here and give a more explicit error if the extension doesn't match?

Reply:
It's checked in the caller, but it certainly won't hurt to do it here (e.g., in case it's called differently in the future)